File: payment-system/src/utils/fee.js
/**
 * A merchant's flat commission "deal" for a currency - see PricingPlan.
 * Real aggregators quote one rate per merchant regardless of which bank
 * issued the card (see CLAUDE.md's "per-merchant deal" decision).
 */
function calculateFee(gross, pricingPlan) {
  const commission = (gross * Number(pricingPlan.commissionPercentage)) / 100;
  const fee = commission + Number(pricingPlan.commissionFixed);
  const net = gross - fee;
  return { fee, net };
}

module.exports = { calculateFee };

File: payment-system/backend/app/core/utils/fee.py
from decimal import Decimal

from app.models import Pos


def calculate_fee(gross: Decimal, pos: Pos) -> tuple[Decimal, Decimal]:
    """Return the gateway fee and merchant net amount for a successful charge."""
    commission = gross * pos.commission_percentage / Decimal("100")
    fee = commission + pos.commission_fixed
    return fee, gross - fee
